File: backend/app/models/mahsulot.py
from sqlalchemy import Boolean, String
from sqlalchemy.orm import Mapped, mapped_column

from app.models.base import Base


class Mahsulot(Base):
    """Fixed ro'yxat: Tola, Lint, Pux, Ulyuk. Kod orqali dasturda ishlatiladi,
    nomi admin panelda ko'rsatiladi."""

    __tablename__ = "mahsulotlar"

    id: Mapped[int] = mapped_column(primary_key=True)
    kod: Mapped[str] = mapped_column(String(20), unique=True)
    nomi: Mapped[str] = mapped_column(String(50))
    faol: Mapped[bool] = mapped_column(Boolean, default=True, server_default="true")

File: frontend/lib/models/mahsulot.dart
class Mahsulot {
  final int id;
  final String kod;
  final String nomi;

  Mahsulot({required this.id, required this.kod, required this.nomi});

  factory Mahsulot.fromJson(Map<String, dynamic> j) => Mahsulot(id: j['id'], kod: j['kod'], nomi: j['nomi']);
}
